### PR TITLE
Adding snippets for Indexes and Transactions.

### PR DESCRIPTION
**Pull request by tmatsuo**

### CODE REVIEW

File: datastore/api/src/functions/concepts.php [hunk 1]
```diff
@@ -725,3 +726,148 @@ function inequality_sort_invalid_not_first(DatastoreClient $datastore)
...
+            'tags' => ['fun', 'programming', 'learn'],
+            'collaborators' => ['alice', 'bob', 'charlie'],
+            'created' => new \DateTime(),
+        ]
+    );
+    // [END exploding_properties]
+    return $task;
+}
+
+// [START transactional_update]
```
Comment (bshaffer) [inline datastore/api/src/functions/concepts.php]: is this intentional to have the whole function for this snippet?

File: datastore/api/src/functions/concepts.php [hunk 2]
```diff
@@ -725,3 +726,148 @@ function inequality_sort_invalid_not_first(DatastoreClient $datastore)
...
+            // if $i >= $retries, the failure is final
+            continue;
+        }
+        // Succeeded!
+        break;
+    }
+    // [END transactional_retry]
+}
+
+// [START transactional_get_or_create]
```
Comment (bshaffer) [inline datastore/api/src/functions/concepts.php]: same here, I feel we should be consistent with having either functions or snippets.

File: datastore/api/src/functions/concepts.php (hunk 1 above)
Comment (tmatsuo) [inline datastore/api/src/functions/concepts.php]: This makes sense in the context of the documentation. In the later section, we will explain how to use this function `transfer_funds`, specifically in the region `transactional_retry`.

File: datastore/api/src/functions/concepts.php (hunk 2 above)
Comment (tmatsuo) [inline datastore/api/src/functions/concepts.php]: Yeah for this snippet, probably we can just show the code, instead of the whole function.

**Comment by tmatsuo** [comment]
@bshaffer Thanks! PTAL

**Comment by bshaffer** [comment]
Awesome, works for me! Well done.

**Pull request merged**